rasa_core/domain.py

```diff
@@ -476,6 +476,20 @@ def persist(self, filename):
         domain_data = self.as_dict()
         utils.dump_obj_as_yaml_to_file(filename, domain_data)
 
+    def persist_clean(self, filename):
+        data = self.as_dict()
+
+        for idx, intent_info in enumerate(data["intents"]):
```

Comment:
this is more of a UX thing, because if we overwrite your nicely formatted domain with the generated one with all the extra stuff, users might not like that. It's not crucial though and I agree it's a bit brittle

Reply:
any suggestions how?